### ISSUE TITLE
FallThrough check doesn't handle synchronized blocks

### ISSUE THREAD

**Issue by cushon** [checkstyle/checkstyle#5037]
Fall through does not occur in this example, but checkstyle reports a warning.

```java
class Test {
  void synchronizedStatement() {
    switch (hashCode()) {
      case 1:
        synchronized (this) {
          break;
        }
      default:
    }
  }
}
```

```
java -jar checkstyle-8.2-all.jar -c google_checks.xml Test.java
...
[WARN] Test.java:8:7: Fall through from previous branch of the switch statement. [FallThrough]
```

Config:
```xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
          "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">


<module name = "Checker">
    <property name="charset" value="UTF-8"/>
    <property name="severity" value="warning"/>
    <property name="haltOnException" value="false"/>
    <module name="TreeWalker">
		<module name="FallThrough"/>
    </module>
</module>

```

**Comment by romani** [comment]
@Vladlis , as you approved this issue, please update description to follow our rules.
config should be provided.

**Comment by Vladlis** [comment]
@romani Updated

**Comment by Vladlis** [comment]
Fix is merged
